src/main/java/org/jenkinsci/plugins/workflow/support/concurrent/Futures.java

```diff
@@ -304,6 +307,28 @@ public void run() {
       return chain(future, wrapperFunction, executor);
     }
 
+    /**
+     * Returns an {@link ExecutorService} to be used as a parameter in other methods.
+     * It calls {@code MoreExecutors#newDirectExecutorService} or falls back to {@code MoreExecutors#sameThreadExecutor}
+     * for compatibility with older (&lt; 18.0) versions of guava.
+     *
+     * @since TODO
+     */
+    public static ExecutorService newExecutorService() {
+        try {
+            try {
+                Method method = MoreExecutors.class.getMethod("newDirectExecutorService");
+                return (ExecutorService) method.invoke(null);
+            } catch (NoSuchMethodException e) {
+                // Guava older than 18, fall back to `sameThreadExecutor`
+                Method method = MoreExecutors.class.getMethod("sameThreadExecutor");
+                return (ExecutorService) method.invoke(null);
```

Comment:
nit: Maybe it would be better to reverse the order until Guava is updated in Jenkins core to avoid throwing `NoSuchMethodException` on a regular basis.

Reply:
done